scarlet-websocket-okhttp/src/main/java/com/tinder/scarlet/websocket/okhttp/OkHttpClientUtils.kt

```diff
@@ -16,5 +18,16 @@ fun OkHttpClient.newWebSocketFactory(requestFactory: RequestFactory): WebSocket.
 }
 
 fun OkHttpClient.newWebSocketFactory(url: String): WebSocket.Factory {
+    if (url.startsWith("ws://")) {
+        try {
+            if (Build.VERSION.SDK_INT >= 23 &&
+                    !NetworkSecurityPolicy.getInstance().isCleartextTrafficPermitted()) {
```

Comment:
we'll need to support the overload method added in 24 as well. see: https://developer.android.com/reference/android/security/NetworkSecurityPolicy.html#isCleartextTrafficPermitted(java.lang.String). but this would need a little refactor to pass in host name.

Reply:
OkHttp has a fairly complex process to get the [host](https://github.com/square/okhttp/blob/4b5615c3323d8abdb875f34a6cd9292825308faf/okhttp/src/main/java/okhttp3/HttpUrl.kt#L1324). But it seems like [this](https://developer.android.com/reference/java/net/URL.html#getHost()) method should suit our purposes.